kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -148,6 +149,38 @@ trait ProcBuilder {
     proc
   }
 
+  def execKillYarnJob(appId: String): Int = {
+    val hadoopHomeOpt = env.get("HADOOP_HOME").orElse {
+      throw KyuubiSQLException(s"Failed to kill yarn job. HADOOP_HOME is not set! " +
+        "For more detail information on installing and configuring Yarn, please visit " +
+        "https://kyuubi.apache.org/docs/stable/deployment/on_yarn.html")
+    }.map { hadoopHome =>
+      Paths.get(hadoopHome, "bin", "yarn").toAbsolutePath.toFile.getCanonicalPath
+    }.get
+    val command = hadoopHomeOpt + " application -kill " + appId
+    val pb = new ProcessBuilder("/bin/sh", "-c", command)
+    var exitValue: Option[Int] = None
+
```

Comment:
we have hadoop-client dependency, keyTab and principle. why we don't use yarn-client to kill this application?

Reply:
Thank you for your suggestion, this is indeed a better solution. I have not implemented it in this way before, I will refine it later or next pr.